compiler/rustc_typeck/src/lib.rs

```diff
@@ -68,6 +68,8 @@ This API is completely unstable and subject to change.
 #![feature(never_type)]
 #![feature(slice_partition_dedup)]
 #![feature(control_flow_enum)]
+#![feature(if_let_guard)]
+#![allow(incomplete_features)]
```

Comment:
Is it okay to use incomplete features in rustc? Based on a quick ripgrep, I can't find any other instances of `allow(incomplete_features)`.

Reply:
Should I remove it?

Reply:
Yes, could you please remove the use of the `if_let_guard` feature? @bors r-